scrapers/traxxx_interface.py

```diff
@@ -0,0 +1,551 @@
+
+    response = self.__callGraphQL(query, variables).get("actors")
+
+    log.info(f'performer traxxxID lookup "{traxxx_performer_id}" returned {len(response)} results')
+
+    if len(response) > 0:
+      return self.parse_performer_to_stash(response[0])
+    else:
+      return None
+
+  def parse_scene_to_stash(self, s):
+    fragment = {}
+
+    fragment["remote_site_id"] = str(s.id)
+
+    if s.title:
+      fragment["title"] = s.title
+
+    if s.description:
+      fragment["details"] = s.description
+
+    if s.url:
+      fragment["url"] = s.url
+
+    if s.date:
+      fragment["date"] = s.date.split("T")[0]
+
+    if s.poster.image:
+      fragment["image"] = s.poster.image
+
+    if s.tags:
+      tags = []
+      for t in s.tags:
+        t = t["tag"]
+        tags.append({
+          "name": t["name"]
+        })
+      fragment["tags"] = tags
+    
+    if s.movies:
+      movies = []
+      for m in s.movies:
+        m = m.movie
+
+
+        movie = {
+          "name": m.title
+        }
+        if m.date:
+          movie["date"] = m.date
+        if m.url:
+          movie["url"] = m.url
+        if m.description:
+          movie["synopsis"] = m.description
+
+        if m.covers:
+          if len(m.covers) >= 1:
+            movie["frontImage"] = m.covers[0]
+          if len(m.covers) >= 2:
+            movie["backImage"] = m.covers[1]
+
+        movies.append(movie)
+      fragment["movies"] = movies
+
+    if s.entity.name:
+      studio = {'name':s.entity.name}
+      if s.entity.url:
+        studio['url'] = s.entity.url
+      fragment["studio"] = studio
+
+    return fragment
+
+  def parse_performer_to_stash(self, p):
+    fragment = {}
+
+    if p.name:
+      fragment["name"] = p.name
+
+    if p.slug:
+      fragment["url"] = f'https://traxxx.me/actor/{p.id}/{p.slug}/'
+
+    if p.gender:
+      fragment["gender"] = p.gender
+
+    if p.birthdate:
+      fragment["birthdate"] = p.birthdate
+
+    if p.dateOfDeath:
+      fragment["death_date"] = p.dateOfDeath
+
+    if p.eyes:
+      fragment["eye_color"] = p.eyes
+
+    if p.hairColor:
+      fragment["hair_color"] = p.hairColor
+
+    if p.heightMetric:
+      fragment["height"] = p.heightMetric
+
+    if p.weightMetric:
+      fragment["weight"] = p.weightMetric
+
+    if p.tattoos:
+      fragment["tattoos"] = p.tattoos
+
+    if p.piercings:
+      fragment["piercings"] = p.piercings
+
+    if p.naturalBoobs == False:
+      fragment["fake_tits"] = "Augmented"
+    if p.naturalBoobs == True:
+      fragment["fake_tits"] = "Natural"
+
+    if all( k in p for k in ['cup','bust','waist','hip'] ):
+      fragment['measurements'] = f"{p['bust']}{p['cup']}-{p['waist']}-{p['hip']}"
+
+    if p.ethnicity:
+      fragment["ethnicity"] = p.ethnicity
+
+    country = p.birthCountry.alpha2
+    if country:
+      fragment["country"] = country
+
+    fragment["images"] = []
+
+    if p.image:
+      fragment["images"].append( p.image )
```

Comment:
From a quick test the first image we might need is the avatar one as the some performers dont have profiles
```py
    if p.avatar:
      fragment["images"].append( p.avatar )
    if p.image:
      fragment["images"].append( p.image )
```

Reply:
Thanks, nice catch, I changed up the wording of the fragments at one point seems I didn't change the avatar image name to image like I did elsewhere so `p.image` never existed resolved it in the fragment